src/index/cache.rs

```diff
@@ -0,0 +1,28 @@
+use crate::id::ResourceId;
+use canonical_path::CanonicalPathBuf;
+use std::str::FromStr;
+
+#[derive(Debug)]
+pub struct IndexCache {
```

Comment:
I don't think this file is useful. You should just dump `ResourceIndex` from `index/mod.rs` into a file, field `collisions` can be ignored while writing to disk. Field `ids` can be ignored as well, only `path2meta` matters. Field `root: PathBuf` is what point to where `.ark` subfolder must be used.

Actually, other fields of `ResourceMeta` should be written as well (initially, I thought the opposite). So, not only `id` but also filesize, modified time etc.

Another direction: while reading `.ark/index` each line `<path>: <id>, <modified>, ...` must create an entry with values of types `CanonicalPathBuf` into `ResourceMeta` in the `ResourceIndex`.